### PR TITLE
GHC.IO.Handle.Lock

### PR DESCRIPTION
**Pull request by phadej**
If we are to use stopgap solution., let's use better one.

### CODE REVIEW

**Comment by hvr** [comment]
Indeed, this one's a much better provisional solution as it avoids introducing a new bug (the #205 regression).

File: hackage-security/src/Hackage/Security/Util/FileLock.hsc [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+--
+-- The modifications were just to the imports and the CPP, since we do not have
+-- access to the HAVE_FLOCK from the ./configure script. We approximate the
+-- lack of HAVE_FLOCK with defined(solaris2_HOST_OS) instead since that is the
+-- only known major Unix platform lacking flock().
+
+import Control.Exception (Exception)
+import Data.Typeable
+
+#if defined(solaris2_HOST_OS)
```
Comment (hvr) [inline hackage-security/src/Hackage/Security/Util/FileLock.hsc]: please add `|| defined(aix_HOST_OS)`

File: hackage-security/src/Hackage/Security/Util/FileLock.hsc [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+-- | Non-blocking version of 'hLock'.
+--
+-- @since 4.10.0.0
+hTryLock :: Handle -> LockMode -> IO Bool
+hTryLock h mode = lockImpl h "hTryLock" mode False
+
+----------------------------------------
+
+#if defined(solaris2_HOST_OS)
```
Comment (hvr) [inline hackage-security/src/Hackage/Security/Util/FileLock.hsc]: here as well

**Comment by phadej** [comment]
aix disabled too.

**Pull request merged**